packages/wallets/src/metamask/pages/elements/networkList.element.ts

```diff
@@ -40,4 +59,82 @@ export class NetworkList {
       await this.networkItemBtn.getByText(chainName).click();
     });
   }
+
+  async openModalNetworkEdit(chainId: any) {
+    await this.dialogSection
+      .getByTestId(`network-list-item-options-button-0x${chainId.toString(16)}`)
+      .click();
+    await this.editNetworkButton.click();
+  }
+
+  async isNetworkExist(
+    networkName: string,
+    rpcUrl: string,
+    chainId: number,
+  ): Promise<boolean> {
+    const existNetworkByName = this.dialogSection.getByTestId(networkName);
+    if (await existNetworkByName.isHidden()) {
+      return false;
+    }
+    // that locator exists only for added networks
+    const elements = this.page.getByTestId(
+      `network-rpc-name-button-0x${chainId.toString(16)}`,
+    );
+    const rpcUrlsFound = await elements.filter({ hasText: rpcUrl }).count();
+    return rpcUrlsFound != 0;
+  }
+
+  async addNetworkManually(
+    networkName: string,
+    networkUrl: string,
+    chainId: number,
+    tokenSymbol: string,
+    blockExplorer = '',
+  ) {
+    await test.step('Open the form to add network manually', async () => {
+      await new Header(this.page).networkListButton.click();
+    });
+
+    if (await this.dialogSection.getByText(networkName).isVisible()) {
+      await this.openModalNetworkEdit(chainId);
+      await this.networkSetting.addRpcForNetwork(networkUrl, blockExplorer);
+    } else {
+      await test.step('Add custom network', async () => {
+        await this.addCustomNetworkButton.click();
+      });
+      await this.networkSetting.addCustomNetwork(
+        networkName,
+        networkUrl,
+        chainId,
+        tokenSymbol,
+        blockExplorer,
+      );
+    }
+  }
+
+  async addPopularNetwork(networkName: string) {
+    await test.step(`Open the form to add the popular network (${networkName})`, async () => {
+      await new Header(this.page).networkListButton.click();
+    });
+    await test.step(`Add the "${networkName}" network`, async () => {
+      await this.dialogSection
+        .getByText(networkName)
+        .locator('../../..')
+        .locator('button:has-text("Add")')
+        .click();
+      // Without awaiting the button is not clickable
+      await this.page.waitForTimeout(500);
+      await new PopoverElements(this.page).approveAddNetworkButton.click();
```

Comment:
Maybe move this locator `approveAddNetworkButton`  to this class?"

Reply:
I think it is an incorrect move

Reply:
why?

Reply:
At the first view, I thought the popover button must locate in the popover class, but that button used only in this process, so I moved it here
Done